include/boost/beast/core/buffers_adaptor.hpp

```diff
@@ -228,8 +229,125 @@ class buffers_adaptor
 
     friend struct buffers_adaptor_test_hook;
 
+
+private:
+
+    friend detail::dynamic_buffer_v2_access;
+
+    auto
+    data_impl(std::size_t pos, std::size_t n)
+    -> mutable_buffers_type
+    {
+        return mutable_buffers_type (begin_, end_,
+            in_pos_ + pos, (std::min)(size(), n));
+    }
+
+    auto
+    data_impl(std::size_t pos, std::size_t n) const
+    -> const_buffers_type
+    {
+        return const_buffers_type (begin_, end_,
+           in_pos_ + pos, (std::min)(size(), n));
+    }
+
+    void
+    shrink_impl(std::size_t n)
+    {
+        boost::ignore_unused(prepare(0));
+
+        if (n == 0)
+            return;
+
+        auto erase_backwards = [this](iter_type i) -> iter_type
+        {
+            auto prev = std::prev(i);
+            end_ = i;
+            return prev;
```

Comment:
This is confusing to me, why not
```
{
  end_ = i;
  return std::prev(i);
}
```
?

Reply:
good question. It probably evolved from something else.